pkgs/tools/networking/p2p/amule/default.nix

```diff
@@ -15,23 +15,10 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "amule-project";
     repo = "amule";
-    rev = version;
```

Comment:
If you want to change to an unstable version, the convention is to set `version` to `unstable-<date-of-the-commit>` to indicate that.

Have you tried juts cherry-picking the necessary commit instead?

Reply:
You should change the commit message accordingly if you decide to stick the unstable version.

Reply:
Did not know the convention, thanks, i'll change accordingly.
I suppose the title of the commit must match that convention as well then.

No, i did not try. It did not cross my mind. 
Plus it sounds like more work than just update to a more "recent" version.

Reply:
> Plus it sounds like more work than just update to a more "recent" version.

It should be as simple as another `fetchpatch`. If the commit that you already identified applies without issues and fixes the problem, that is preferable. We do not know the exact state of master, its better to stick to upstream releases if possible.

If its not that easy, I wouldn't put too much work into it either though. Then an update is fine as well.

Reply:
Although in this case arch linux apparently also ships a master release, so that's probably alright as well. In any case the libupnp commit should be removed here.

Reply:
> If its not that easy, I wouldn't put too much work into it either though. Then an update is fine as well.

i just learn about this with the other diff on gmrender-resurrect ;)

> If its not that easy, I wouldn't put too much work into it either though. Then an update is fine as well.

but here it's not that easy indeed, the patch does not apply easily.
```
...
Hunk #1 FAILED at 33.
...
```

An update it is then ;)